tests/libvmi/cloudinit.go

```diff
@@ -60,6 +62,30 @@ func WithCloudInitNoCloudNetworkData(data string) Option {
 	}
 }
 
+// WithCloudInitNoCloudEncodedNetworkData adds cloud-init no-cloud base64 encoded network data.
+func WithCloudInitNoCloudEncodedNetworkData(networkData string) Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		addDiskVolumeWithCloudInitNoCloud(vmi, cloudInitDiskName, v1.DiskBusVirtio)
+
+		volume := getVolume(vmi, cloudInitDiskName)
+		volume.CloudInitNoCloud.NetworkDataBase64 = base64.StdEncoding.EncodeToString([]byte(networkData))
+		volume.CloudInitNoCloud.NetworkData = ""
+
+	}
+}
+
+// WithCloudInitNoCloudNetworkDataSecret adds cloud-init no-cloud network data from secret.
+func WithCloudInitNoCloudNetworkDataSecret(secretName string) Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		addDiskVolumeWithCloudInitNoCloud(vmi, cloudInitDiskName, v1.DiskBusVirtio)
+
+		volume := getVolume(vmi, cloudInitDiskName)
+		volume.CloudInitNoCloud.NetworkDataSecretRef = &k8scorev1.LocalObjectReference{Name: secretName}
```

Comment:
What about passing in `NetworkDataSecretRef` value as the input?.

Reply:
It wont do because it would be mentioned in the function signature and introduce dependency on `k8s.io/api/core/v1`

Reply:
It is explicit about what you change with a type check. Not related to the dependency. But if you prefer it like this I do not mind much.

Reply:
I tried to make it easy to work with in tests, the test shouldn't care for this wrapping object(`LocalObjectRefrence`1), just the secret object name it that it creates.

Yeah, I prefer to leave it as is.